Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp

```diff
@@ -1048,6 +1048,77 @@ static gboolean webkitWebViewAccumulatorObjectHandled(GSignalInvocationHint*, GV
     return !object;
 }
 
+#if PLATFORM(GTK) && USE(GTK4)
+static GdkEvent* gValueGetEvent(const GValue* value)
+{
+    g_return_val_if_fail(G_VALUE_HOLDS(value, GDK_TYPE_EVENT), NULL);
+
+    return reinterpret_cast<GdkEvent*>(value->data[0].v_pointer);
+}
+
+typedef gboolean (*ContextMenuCallback) (gpointer, WebKitContextMenu*, GdkEvent*, WebKitHitTestResult*, gpointer);
+
+static void webkitWebViewContextMenuMarshal(GClosure* closure, GValue* returnValue, guint nParams, const GValue* params, gpointer, gpointer marshalData)
+{
+    GCClosure* cc = (GCClosure *) closure;
+    gpointer data1, data2;
+
+    g_return_if_fail(returnValue);
+    g_return_if_fail(nParams == 4);
+
+    if (G_CCLOSURE_SWAP_DATA(closure)) {
+        data1 = closure->data;
+        data2 = g_value_peek_pointer(&params[0]);
+    } else {
+        data1 = g_value_peek_pointer(&params[0]);
+        data2 = closure->data;
+    }
+
+    auto* menu = WEBKIT_CONTEXT_MENU(g_value_get_object(&params[1]));
+    auto* event = gValueGetEvent(&params[2]);
+    auto* result = WEBKIT_HIT_TEST_RESULT(g_value_get_object(&params[3]));
+
+    ContextMenuCallback callback = (ContextMenuCallback) (marshalData ? marshalData : cc->callback);
+    gboolean ret = callback(data1, menu, event, result, data2);
+
+    g_value_set_boolean(returnValue, ret);
+}
+
+static void webkitWebViewContextMenuMarshalVa(GClosure *closure, GValue *returnValue, gpointer instance, va_list args, gpointer marshalData, int, GType*)
+{
+    GCClosure* cc = (GCClosure *) closure;
+    gpointer data1, data2;
+
+    g_return_if_fail(returnValue);
+
+    if (G_CCLOSURE_SWAP_DATA(closure)) {
+        data1 = closure->data;
+        data2 = instance;
+    } else {
+        data1 = instance;
+        data2 = closure->data;
+    }
+
+    va_list argsCopy;
+    G_VA_COPY(argsCopy, args);
+
+    auto* menu = WEBKIT_CONTEXT_MENU(g_object_ref(va_arg(argsCopy, gpointer)));
+    auto* event = gdk_event_ref(reinterpret_cast<GdkEvent*>(va_arg(argsCopy, gpointer)));
+    auto* result = WEBKIT_HIT_TEST_RESULT(g_object_ref(va_arg(argsCopy, gpointer)));
```

Comment:
Ownership via raw pointers is prohibited. Need to use the smart pointers, e.g.:

```suggestion
    GRefPtr<WebKitContextMenu> menu = WEBKIT_CONTEXT_MENU(va_arg(argsCopy, gpointer));
    GRefPtr<GdkEvent> event = reinterpret_cast<GdkEvent*>(va_arg(argsCopy, gpointer));
    GRefPtr<WebKitHitTestResult> result = WEBKIT_HIT_TEST_RESULT(va_arg(argsCopy, gpointer));
```

Reply:
GRefPtr doesn't work with GdkEvent atm.

Reply:
You get to fix that then. Should be very easy. Just copy/paste in GRefPtrGtk.h and GRefPtrGtk.cpp and have it use gdk_event_ref/unref.